src/Doctrine/Migration/MigrationService.php

```diff
@@ -206,17 +207,22 @@ public function generateDiffSqls(): array
     {
         $schemaTool = new SchemaTool($this->entityManager);
         $platform = $this->entityManager->getConnection()->getDatabasePlatform();
-        $classes = $this->entityManager->getMetadataFactory()->getAllMetadata();
+        $classMetadata = $this->entityManager->getMetadataFactory()->getAllMetadata();
 
         $schemaManager = $this->entityManager->getConnection()->getSchemaManager();
         $fromSchema = $schemaManager->createSchema();
-        $toSchema = $schemaTool->getSchemaFromMetadata($classes);
+        $toSchema = $schemaTool->getSchemaFromMetadata($classMetadata);
 
         $this->excludeTablesFromSchema($fromSchema);
         $this->excludeTablesFromSchema($toSchema);
 
-        $schemaComparator = new Comparator();
-        $schemaDiff = $schemaComparator->compare($fromSchema, $toSchema);
+        if (method_exists($schemaManager, 'createComparator')) {
+            $schemaComparator = $schemaManager->createComparator();
+        } else {
+            $schemaComparator = new Comparator();
+        }
```

Comment:
Try to figure out how to deal with supporting multiple doctrine versions. If everything fails, you can do conditional ignore like this: https://github.com/shipmonk-rnd/phpstan-rules/blob/master/phpstan.ignores.php